crates/holochain/src/core/ribosome/host_fn/block_agent.rs

```diff
@@ -43,23 +58,102 @@ mod test {
+        let _response0: ZomeCallResponse = conductor
+            .call(&alice, "try_cap_claim", cap_for.clone())
+            .await;
+        let _response1: ZomeCallResponse = conductor
+            .call(&alice, "try_cap_claim", cap_for.clone())
+            .await;
 
-        let _: () = conductor.call(&bob, "block_agent", alice_pubkey.clone()).await;
+        let _: () = conductor
+            .call(&bob, "block_agent", alice_pubkey.clone())
+            .await;
 
-        let response2: ConductorApiResult<ZomeCallResponse> = conductor.call_fallible(&alice, "try_cap_claim", cap_for.clone()).await;
+        let response2: ConductorApiResult<ZomeCallResponse> = conductor
+            .call_fallible(&alice, "try_cap_claim", cap_for.clone())
+            .await;
         assert!(response2.is_err());
 
         let _: () = conductor.call(&bob, "unblock_agent", alice_pubkey).await;
 
         let _response3: ZomeCallResponse = conductor.call(&alice, "try_cap_claim", cap_for).await;
     }
 
+    #[tokio::test(flavor = "multi_thread")]
+    async fn zome_call_get_block() {
+        holochain_trace::test_run().ok();
+
+        let (dna_file, _, _) = SweetDnaFile::unique_from_test_wasms(vec![TestWasm::Create]).await;
+
+        let config = SweetConductorConfig::standard().tune(|tune| {
+            tune.gossip_peer_on_success_next_gossip_delay_ms = 1000;
+            tune.gossip_peer_on_error_next_gossip_delay_ms = 1000;
+            tune.gossip_round_timeout_ms = 3000;
+        });
+        let mut conductors = SweetConductorBatch::from_config(3, config).await;
+        let apps = conductors
+            .setup_app("create", &[dna_file.clone()])
+            .await
+            .unwrap();
+
+        let ((alice_cell,), (bob_cell,), (carol_cell,)) = apps.into_tuples();
+
+        let alice = alice_cell.zome(TestWasm::Create);
+        let bob = bob_cell.zome(TestWasm::Create);
+
+        let bob_pubkey = bob_cell.cell_id().agent_pubkey();
+
+        conductors.reveal_peer_info(0, 1).await;
+        conductors.reveal_peer_info(1, 0).await;
+
+        let alice_conductor = conductors.get(0).unwrap();
+        let bob_conductor = conductors.get(1).unwrap();
+
+        let action0: ActionHash = alice_conductor.call(&alice, "create_entry", ()).await;
+
+        consistency_10s([&alice_cell, &bob_cell]).await;
+
+        // Before bob is blocked he can get posts just fine.
+        let bob_get0: Option<Record> = bob_conductor.call(&bob, "get_post", action0).await;
+        // Await bob's init to propagate to alice.
+        consistency_10s([&alice_cell, &bob_cell]).await;
+        assert!(bob_get0.is_some());
+
+        // Bob gets blocked by alice.
+        let _block: () = alice_conductor
+            .call(&alice, "block_agent", bob_pubkey)
+            .await;
+
+        let action1: ActionHash = alice_conductor.call(&alice, "create_entry", ()).await;
```

Comment:
Should bob be able to still get entry0 now? I suppose not. So you could omit creating the second entry.